straxen/itp_map.py

```diff
@@ -206,14 +207,16 @@ def _regular_grid_interpolator(csys, map_data, array_valued, **kwargs):
             map_data = map_data.reshape(*grid_shape)
 
         config = dict(bounds_error=False, fill_value=None)
+        kwargs = straxen.filter_kwargs(RegularGridInterpolator, kwargs)
         config.update(kwargs)
+        
         return RegularGridInterpolator(tuple(grid), map_data, **config)
 
     @staticmethod
     def _weighted_nearest_neighbors(csys, map_data, array_valued, **kwargs):
         if array_valued:
             map_data = map_data.reshape((-1, map_data.shape[-1]))
-
+        kwargs = straxen.filter_kwargs(InterpolateAndExtrapolate, kwargs)
```

Comment:
This looks strongly like a decorator to me. Any reasons to not have it as one?

Reply:
that is a great idea!